docs/antrea-l7-network-policy.md

```diff
@@ -201,6 +203,91 @@ spec:
         - http: {}       # automatically dropped, and subsequent rules will not be considered.
 ```
 
+### TLS
+
+A typical layer 7 NetworkPolicy TLS protocol is as below:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: NetworkPolicy
+metadata:
+  name: ingress-allow-tls-request-to-api-v2
+spec:
+  priority: 5
+  tier: application
+  appliedTo:
+    - podSelector:
+        matchLabels:
+          app: web
+  ingress:
+    - name: allow-http   # Allow inbound TLS requests to TLS server name "foo.bar.com" from Pods with app=client label.
+      action: Allow      # All other traffic from these Pods will be automatically dropped, and subsequent rules will not be considered.
+      from:
+        - podSelector:
+            matchLabels:
+              app: client
+      l7Protocols:
+        - tls:
+            sni: "foo.bar.com"
+    - name: drop-other   # Drop all other inbound traffic (i.e., from Pods without the app=client label or from external clients).
+      action: Drop
+```
+
+**sni**: The `sni` field represents the service name present in TLS header to match. Both exact matches and wildcards are
+supported, e.g. `*.foo.com`, `*.foo.*`, `foo.bar.com`. If not set, the rule matches any service name.
+
+#### More examples TLS
+
+The following NetworkPolicy prevents applications from accessing unauthorized TLS server names:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: allow-web-tls-access-to-internal-domain
+spec:
+  priority: 5
+  tier: securityops
+  appliedTo:
+    - podSelector:
+        matchLabels:
+          egress-restriction: internal-domain-only
+  egress:
+    - name: allow-dns          # Allow outbound DNS requests.
+      action: Allow
+      ports:
+        - protocol: TCP
+          port: 53
+        - protocol: UDP
+          port: 53
+    - name: allow-tls-only    # Allow outbound TLS requests towards foo.bar.com. 
+      action: Allow           # As the rule's "to" and "ports" are empty, which means it selects traffic to any network
+      l7Protocols:            # peer's any port using any transport protocol, all outbound TLS requests towards other
+        - tls:                # service names and non-TLS requests will be automatically dropped, and subsequent rules
```

Comment:
not sure why you use "service name" in several places. It should always be "server name".

Reply:
My fault. It should be "server name".